src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -123,6 +147,30 @@ internal static ObsoleteDiagnosticKind GetObsoleteDiagnosticKind(Symbol symbol,
                     // later stage
                     return ObsoleteDiagnosticKind.LazyPotentiallySuppressed;
             }
+
+            static bool isExperimentalSymbol(NamedTypeSymbol namedTypeSymbol)
+            {
+                return namedTypeSymbol.Arity == 0
+                    && namedTypeSymbol.HasNameQualifier("System.Diagnostics.CodeAnalysis")
```

Comment:
>namedTypeSymbol.HasNameQualifier("System.Diagnostics.CodeAnalysis") [](http://example.com/codeflow?start=23&length=67)

Perhaps `namedTypeSymbol.IsWellKnownDiagnosticsCodeAnalysisTopLevelType()` since that method doesn't require splitting strings, and also checks the type is a top-level type in that namespace.